tools/azure-sdk-tools/devtools_testutils/proxy_startup.py

```diff
@@ -57,6 +55,29 @@ def get_image_tag() -> str:
     return image_tag
 
 
+def ascend_to_root(start_dir_or_file: str) -> str:
+    """
+    Given a path, ascend until encountering a folder with a `.git` folder present within it. Return that directory.
+
+    :param str start_dir_or_file: The starting directory or file. Either is acceptable.
+    """
+    if os.path.isfile(start_dir_or_file):
+        current_dir = os.path.dirname(start_dir_or_file)
+    else:
+        current_dir = start_dir_or_file
+
+    while current_dir is not None and not (os.path.dirname(current_dir) == current_dir):
+        possible_root = os.path.join(current_dir, ".git")
+
+        # we need the git check to prevent ascending out of the repo
+        if os.path.exists(possible_root):
+            return current_dir
+        else:
+            current_dir = os.path.dirname(current_dir)
+
+    raise Exception('Requested target "{}" does not exist within a git repo.'.format(start_dir_or_file))
```

Comment:
nit: I love f-strings
```suggestion
    raise Exception(f'Requested target "{start_dir_or_file}" does not exist within a git repo.')
```

Reply:
Know what. you're absolutely right. I should modernize my style on this.